### PR TITLE
Break complex variants

### PR DESCRIPTION
**Pull request by Fohlen**
The following PR adds the `vcfbreakcomplexvariants` command. We use our own [needleman-wunsch implementation](https://github.com/imgag/needleman-wunsch). Tests are shamelessly copied from [vt](https://github.com/atks/vt/tree/master/test/decompose_blocksub).

Currently the stats output breaks the program, I'll figure out soon why that is 👍

### CODE REVIEW

File: src/VcfBreakComplexVariants/main.cpp [hunk 1]
```diff
@@ -0,0 +1,252 @@
+#include <algorithm>
+#include <vector>
+#include <string>
+#include <QFile>
+#include "ToolBase.h"
+#include "Exceptions.h"
+#include "Helper.h"
+#include "VcfFile.h"
+#include "../../needleman-wunsch/NeedlemanWunsch.hpp" // TODO: This does seem like an odd import path
```
Comment (Fohlen) [inline src/VcfBreakComplexVariants/main.cpp]: @marc-sturm how can I fix the include here?

File: src/VcfBreakComplexVariants/main.cpp [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+		while(!in_p->atEnd())
+		{
+			QByteArray line = in_p->readLine();
+			// Skip empty lines and headers
+			if (line.trimmed().isEmpty() || line.startsWith("#"))
+			{
+				out_p->write(line);
+				if (!inserted_info) // Insert right after version line
+				{
+					out_p->write("##INFO=<ID=OLD_CLUMPED,Number=1,Type=String,Description=\"Original chr:pos:ref|alt encoding\">\n");
```
Comment (marc-sturm) [inline src/VcfBreakComplexVariants/main.cpp]: rename: BBC > Comment

File: src/VcfBreakComplexVariants/main.cpp [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+	/**
+	 * @brief classifyVariant
+	 * Classifies variants according to https://genome.sph.umich.edu/wiki/Variant_classification
+	 * Assumes that both REF and ALT are already trimmed (no whitespaces, tabs)
+	 *
+	 * @param ref - the reference sequence
+	 * @param alt - the alternating sequence
+	 * @return VariantType
+	 */
+	VariantType classifyVariant(const QByteArray& ref, const QByteArray& alt)
```
Comment (marc-sturm) [inline src/VcfBreakComplexVariants/main.cpp]: Move to VcfFile:
- CLUMPED > COMPLEX
- remove min(...)

File: src/VcfBreakComplexVariants/main.cpp [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+	}
+
+	/**
+	 * @brief editDistance
+	 * Returns the levensthein distance for two sequences
+	 * @param a
+	 * @param b
+	 * @return
+	 */
+	static unsigned int editDistance(const QByteArray& a, const QByteArray& b)
```
Comment (marc-sturm) [inline src/VcfBreakComplexVariants/main.cpp]: - Move to NGSHelper
- rename to levenstein

File: src/VcfBreakComplexVariants/main.cpp [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+					d[i - 1][j - 1] + (a[i - 1] == b[j - 1] ? 0 : 1)
+				});
+			}
+		}
+
+		return d[len_a][len_b];
+	}
+
+	//Returns the content of a column by index (tab-separated line)
+	static QByteArray getPartByColumn(const QByteArray& line, int index)
```
Comment (marc-sturm) [inline src/VcfBreakComplexVariants/main.cpp]: Move to VcfFile class

**Pull request merged**